mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1818,6 +1824,23 @@ def _log_eval_table(self):
             uri=mlflow.get_artifact_uri(artifact_file_name)
         )
 
+    def _log_genai_custom_metrics(self, genai_custom_metrics):
+        if len(genai_custom_metrics) == 0:
+            return
+
+        data = {"name": [], "version": [], "metric_args": []}
+        for genai_metric_args in genai_custom_metrics:
+            data["name"].append(genai_metric_args["name"])
+            # Custom metrics created from make_genai_metric_from_prompt don't have version
+            data["version"].append(genai_metric_args.get("version", None))
+            data["metric_args"].append(genai_metric_args)
+        mlflow.log_table(data, artifact_file=_GENAI_CUSTOM_METRICS_FILE_NAME)
+
+        name = _GENAI_CUSTOM_METRICS_FILE_NAME.split(".", 1)[0]
+        self.artifacts[name] = JsonEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(_GENAI_CUSTOM_METRICS_FILE_NAME)
+        )
+
```

Comment:
This might be a bit more extensible by leaving the list definitions (the reason why they need to be defined externally to begin with) at their base level and then constructing the final structure after the loop:

```suggestion
    def _log_genai_custom_metrics(self, genai_custom_metrics):
        if not genai_custom_metrics:
            return

        names = []
        versions = []
        metric_args_list = []

        for metric_args in genai_custom_metrics:
            names.append(metric_args["name"])
            versions.append(metric_args.get("version", None))
            metric_args_list.append(metric_args)

        data = {
            "name": names,
            "version": versions,
            "metric_args": metric_args_list
        }

        mlflow.log_table(data, artifact_file=_GENAI_CUSTOM_METRICS_FILE_NAME)

        artifact_name = _GENAI_CUSTOM_METRICS_FILE_NAME.rsplit(".", 1)[0]
        self.artifacts[artifact_name] = JsonEvaluationArtifact(
            uri=mlflow.get_artifact_uri(_GENAI_CUSTOM_METRICS_FILE_NAME)
        )

```

Reply:
that makes sense. have incorporated the feedback